### PR TITLE
7pm-2-MiniEpic-SearchHistory

### PR DESCRIPTION
**Pull request by bowenduanmu**
Combine the results of three substories and modified SearchHistory controller for different situation of admin users and member users. Check the function of page to correctly show the searches result table.

### CODE REVIEW

File: src/main/java/edu/ucsb/mapache/controllers/SearchHistoryController.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            throws JsonProcessingException {
+        if (!authControllerAdvice.getIsMember(authorization))
+            return getUnauthorizedResponse("member");
+        
+       if (!authControllerAdvice.getIsAdmin(authorization)) {
+            DecodedJWT jwt = getJWT(authorization);
+            Map<String, Object> customClaims = jwt.getClaim(propertiesService.getNamespace()).asMap();
+            String firstName = (String) customClaims.get("given_name");
+            String lastName = (String) customClaims.get("family_name");
+            String userid = firstName.concat(lastName);
```
Comment (pconrad) [inline src/main/java/edu/ucsb/mapache/controllers/SearchHistoryController.java:71]: These lines should be covered by tests.

File: src/test/java/edu/ucsb/mapache/controllers/SearchHistoryControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    // public ResponseEntity<String> getSearches(@RequestHeader("Authorization")
+    // String authorization)
+
+    // I wrote 'case 2' because you'll probably need a few cases, e.g.
+    // for not logged in, logged in as admin, logged in as member, etc.
+    // depending on what the code is checking for
+  }
+
+  // ETC... until you've tested all of the cases for all of the methods.
+  // Look in other controller tests for examples
```
Comment (pconrad) [inline src/test/java/edu/ucsb/mapache/controllers/SearchHistoryControllerTests.java:143]: This code is NOT in final form to merge.   This is still essentially a draft PR.

Please let us know when it is ready to merge.  Code that's ready to merge doesn't have big blocks of commented out code in it.  This is code that I wrote to help your team... and you haven't done anything with it yet, at least not in this branch.

File: javascript/src/main/App.js [hunk 3]
```diff
@@ -46,10 +46,10 @@ function App() {
           <AuthorizedRoute path="/admin/searchedTerms" exact component={SearchedTerms} authorizedRoles={["admin"]} />
           <AuthorizedRoute path="/member/channels" exact component={ChannelList} authorizedRoles={["admin","member"]} />
           <AuthorizedRoute path="/member/search" exact component={Search} authorizedRoles={["admin","member"]} />
+          <AuthorizedRoute path="/member/searchhistory" exact component={SearchHistory} authorizedRoles={["admin","member"]} />
           <AuthorizedRoute path="/member/messages/search" component={SearchResults} authorizedRoles={["admin","member"]} />
           <AuthorizedRoute path="/member/analyzemessages/reactions" exact component={AnalyzeReactions} authorizedRoles={["admin", "member"]} />
           <AuthorizedRoute path="/member/listViewChannels/:channel" component={ChannelPageList} authorizedRoles={["admin","member"]} />
-          <AuthorizedRoute path="/member/channels/:channel/links" component={ChannelPageLinks} authorizedRoles={["admin","member"]} />
```
Comment (pconrad) [inline javascript/src/main/App.js:52]: Why did you remove this route that appears to have nothing to do with your PR?

You should put it back unless you have an explanation for why you are removing it.

**Review by pconrad — CHANGES_REQUESTED** [review]
This seems to be incomplete.

**Comment by bowenduanmu** [comment]
This PR is included in another PR.

**Pull request closed without merging**